### ISSUE TITLE
Accessibility: Accordions missing header markup

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#51796]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The accordions in the Order Summary, Add a coupon sections do not have heading tags wrapped around the buttons that open and close them. This may make the content contained in each accordion less clear for screen reader users.   Additionally, without heading markup, blind users will not be able to quickly navigate to each accordion using the headings list in their screen reader.

![T001_SR_HeadingNotAnnounced](https://github.com/user-attachments/assets/f7932c7d-6b3c-4850-9181-6b01051e5940)

### Component or Page: 

**Page: Checkout Process (Block-Based)**

### Name affected element:

1. Order summary
2. Add a coupon

### Relevant code snippets:

`<button aria-expanded="false" class="wc-block-components-panel__button"><svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" width="24" height="24" aria-hidden="true" class="wc-block-components-panel__button-icon" focusable="false"><path d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"></path></svg><span class="wc-block-components-order-summary__button-text">Order summary</span></button>`

### Recommended fix:

The text that serves as the control for opening and closing the accordion should be marked appropriately to indicate it is a subheading in the context of the page. This heading tag should be wrapped around the button tag that opens and closes the accordion. In this case the accordion headings should be H2s.
 
 For additional information on accordion accessibility expectations see https://equalizedigital.com/how-to-code-accessible-accordions/

**Comment by Manussakis** [comment]
@woocommerce/rubik , Could you please give me more context on how to make the Order summary collapsible on the desktop, like the screenshot in the issue description?

I was only able to see it collapsible on mobile and it's still different from the screenshot provided:
![Image](https://github.com/user-attachments/assets/6bd37d85-01fa-497e-a4be-f31a2c46b0e5)

cc: @vikrampm1

**Comment by samueljseay** [comment]
@Manussakis we recently made some changes in this area. It is now rendered twice on mobile once top of screen and once at bottom, with only the top one collapsible, and it is not collapsible in desktop view.

**Comment by Manussakis** [comment]
I set this task as blocked since we are waiting for Seghir Nadir's response to [this question](https://github.com/woocommerce/woocommerce/pull/52844#issuecomment-2477557684) in the PR. 

cc: @vikrampm1
